src/components/DownloadLink/DownloadLink.tsx

```diff
@@ -0,0 +1,33 @@
+import Link, { DOWNLOAD_LINK_VARIANTS } from '../Link';
+
+export interface DownloadLinkProps extends React.HTMLAttributes<HTMLAnchorElement> {
+  /**
+   * Optionally customize the name of the downloaded file. If `href` is "original.jpg" and `filename` is "custom-name", the file will be downloaded as "custom-name.jpg".
+   */
+  filename?: string;
+  /**
+   * The URL of the file to download. Must be on the same origin as the current page.
+   * @example
+   * ✅ "https://www.bostonscientific.com/image.jpg"
+   * ✅ "/image.jpg"
+   * ❌ "https://www.google.com/image.jpg"
+   */
+  href: string;
```

Comment:
Changed "source" to "href" to match other link patterns. It is required here, though, unlike in the Link component.